tools/ami-cleanup/src/internal/cleanup_test.go

```diff
@@ -0,0 +1,667 @@
+	usWest2 := "us-west-2"
+	expectedRegions := []accountTypes.Region{
+		{
+			RegionName:      &usEast1,
+			RegionOptStatus: accountTypes.RegionOptStatusEnabled,
+		},
+		{
+			RegionName:      &usEast2,
+			RegionOptStatus: accountTypes.RegionOptStatusEnabledByDefault,
+		},
+		{
+			RegionName:      &usWest1,
+			RegionOptStatus: accountTypes.RegionOptStatusEnabledByDefault,
+		},
+		{
+			RegionName:      &usWest2,
+			RegionOptStatus: accountTypes.RegionOptStatusEnabled,
+		},
+	}
+
+	tests := []struct {
+		desc            string
+		mockListRegions func(ctx context.Context, params *account.ListRegionsInput, optFns ...func(*account.Options)) (*account.ListRegionsOutput, error)
+		shouldError     bool
+		expectedRegions []accountTypes.Region
+	}{
+		{
+			desc: "fail if API call errors",
+			mockListRegions: func(ctx context.Context, params *account.ListRegionsInput, optFns ...func(*account.Options)) (*account.ListRegionsOutput, error) {
+				return nil, errors.New("some API call error")
+			},
+			shouldError: true,
+		},
+		{
+			desc: "no error when API call does not error",
+			mockListRegions: func(ctx context.Context, params *account.ListRegionsInput, optFns ...func(*account.Options)) (*account.ListRegionsOutput, error) {
+				return &account.ListRegionsOutput{}, nil
+			},
+			shouldError: false,
+		},
+		{
+			desc: "only enabled regions requested",
+			mockListRegions: func(ctx context.Context, params *account.ListRegionsInput, optFns ...func(*account.Options)) (*account.ListRegionsOutput, error) {
+				enabledStatuses := []accountTypes.RegionOptStatus{accountTypes.RegionOptStatusEnabled, accountTypes.RegionOptStatusEnabledByDefault}
+				slices.Sort(enabledStatuses)
+				slices.Sort(params.RegionOptStatusContains)
+				if slices.Compare(enabledStatuses, params.RegionOptStatusContains) != 0 {
+					return nil, trace.Errorf("the requested region statuses %#v did not match the expected region statuses %#v", params.RegionOptStatusContains, enabledStatuses)
+				}
+				return &account.ListRegionsOutput{}, nil
+			},
+			shouldError: false,
+		},
+		{
+			desc: "all results are returned",
+			mockListRegions: func(ctx context.Context, params *account.ListRegionsInput, optFns ...func(*account.Options)) (*account.ListRegionsOutput, error) {
+				return &account.ListRegionsOutput{
+					Regions:   expectedRegions,
+					NextToken: nil,
+				}, nil
+			},
+			expectedRegions: expectedRegions,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			// Setup the application instance
+			application := &ApplicationInstance{
+				accountClientGenerator: func(cfg *aws.Config) IAccountApi {
+					return &MockAccountAPI{
+						MockListRegions: test.mockListRegions,
+					}
+				},
+			}
+
+			// Run the function under test
+			regions, err := application.getEnabledRegions(context.Background(), application.accountClientGenerator(nil))
+
+			// Verify the results
+			checkError(t, test.shouldError, err)
```

Comment:
Change `shouldError` from a bool to a `require.ErrorAssertionFunc`, and for each test case instead of passing in true/false, pass in `require.Error` or `require.NoError`. I'd rename it from `shouldError` to `checkError` too.

When that's done, you can remove your `checkError` function completely, and just run `test.checkError(t, err)`.